ui/lib/services/bom_bar_client.dart

```diff
@@ -38,9 +38,10 @@ class BomBarClient {
     ));
   }
 
-  /// Returns all available projects.
-  Future<List<Project>> getProjects() async {
-    final response = await dio.getUri(_projectsUrl);
+  /// Returns all projects or projects matching the provided [fragment] fragment.
+  Future<List<Project>> findProjectsBySearchFragment([String? fragment]) async {
```

Comment:
Wondering if we should also rename this method to `findProjects(...)`, as the search fragment seems obvious from the parameter. (And if no fragment is provided, it still finds project.)